addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/handler/ChromecastHandler.java

```diff
@@ -342,36 +425,74 @@ private void handleCcMediaStatus(final MediaStatus mediaStatus) {
                         PlayPauseType.PLAY);
                 break;
             default:
+                logger.debug("Unknown mediaStatus: {}", mediaStatus.playerState);
                 break;
         }
+
+        updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_CURRENT_TIME),
+                new DecimalType(mediaStatus.currentTime));
+
+        updateMediaInfoStatus(mediaStatus.media);
     }
 
-    private void handleCcVolume(final Volume volume) {
-        PercentType value = new PercentType((int) (volume.level * 100));
-        updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_VOLUME), value);
-        this.volume = value;
-        updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_MUTE),
-                volume.muted ? OnOffType.ON : OnOffType.OFF);
+    private void updateMediaInfoStatus(final Media media) {
+        Double duration = 0.0;
+        String metadataType = Media.MetadataType.GENERIC.name();
+
+        if (media != null) {
+            duration = media.duration;
+            metadataType = media.getMetadataType().name();
+        }
+
+        updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_DURATION),
+                duration == null ? UnDefType.UNDEF : new DecimalType(duration));
+        updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_METADATA_TYPE),
+                new StringType(metadataType));
+
+        updateMetadataStatus(media == null ? Collections.emptyMap() : media.metadata);
```

Comment:
if media == null of media.metadata is null you're `wasting` time in that function, should you even call it then? Because calling get on that map will return null for each call so a lot of null's to handle. Or do you want to be sure that split out a lot of UNDEF's?